main.py

```diff
@@ -23,28 +37,80 @@ def image_generator(bbox_list, time_interval, evalscript, maxcc):
 
 
 def main():
-    ssl._create_default_https_context = (
-        ssl._create_unverified_context
-    )  # fix for SSL error on Mac
-
     bbox = BBox(MANILLA_BAY_BBOX, crs=CRS.WGS84)
     time_interval = ("2023-08-01", "2023-09-01")
     maxcc = 0.5
     out_dir = "images"
+
+    ssl._create_default_https_context = (
+        ssl._create_unverified_context
+    )  # fix for SSL error on Mac
+
     bbox_list = UtmZoneSplitter([bbox], crs=CRS.WGS84, bbox_size=5000).get_bbox_list()
 
     data_gen = image_generator(bbox_list, time_interval, L2A_12_BANDS, maxcc)
+    detector = SegmentationModel.load_from_checkpoint(
+        CHECKPOINTS["unet++1"], map_location="cpu", trust_repo=True
+    )
+    predictor = ScenePredictor(device="cpu")
 
     for data in data_gen:
-        detector = SegmentationModel.load_from_checkpoint(
-            CHECKPOINTS["unet++1"], map_location="cpu", trust_repo=True
-        )
         for _d in data:
             if _d.content is not None:
-                predictor = ScenePredictor(device="cpu")
```

Comment:
Maybe we can add a check in here whether the prediction already exists in the db. I was thinking if you start making predictions for a specific area and it stops in the middle then you'd have to rerun everything, would be faster if we can skip the calculating the predictions in this case? What do you think?
(But I think that would be a new task, I'd first close this pull request)